label_sync/main.go

```diff
@@ -186,22 +198,43 @@ func validate(labels []Label, parent string, seen map[string]string) error {
 	return nil
 }
 
+func copyStringMap(originalMap map[string]string) map[string]string {
+	newMap := make(map[string]string)
+	for k, v := range originalMap {
+		newMap[k] = v
+	}
+	return newMap
+}
+
 // Ensures the config does not duplicate label names
 func (c Configuration) validate() error {
+	// Check default labels
 	seen := make(map[string]string)
-	if err := validate(c.Labels, "", seen); err != nil {
+	if err := validate(c.Default, "default", seen); err != nil {
 		return fmt.Errorf("invalid config: %v", err)
 	}
+	// Check other repos labels
+	for repo, labels := range c.Repos {
+		// Duplicate seen to avoid modifications
+		defaultLabels := copyStringMap(seen)
+		// Validate as usual
+		// Will complain if a label is both in default and repo
+		if err := validate(labels, repo, defaultLabels); err != nil {
+			return fmt.Errorf("invalid config: %v", err)
+		}
+	}
 	return nil
 }
 
 // LabelsByTarget returns labels that have a given target
-func (c Configuration) LabelsByTarget(target LabelTarget) (labels []Label) {
-	for _, label := range c.Labels {
+func LabelsByTarget(labels []Label, target LabelTarget) (filteredLabels []Label) {
```

Comment:
`LabelsByTarget` sounds like a mapping will be returned, maybe `LabelsForTarget`?